### DISCUSSION TITLE
How to Permanently Show the Slug Input in Panel?

### DISCUSSION THREAD

**Discussion by ghost** [mecha-cms/mecha#130] [Q&A] [answered]
Halo, selamat malam Mas Taufik @taufik-nurrohman.

Saya mau tanya soal slug.

Kenapa pengaturan isian slug dalam *page* di panel baru muncul setelah kita mempublikasikannya?

Apa alasannya tidak membuat input slug sejak awal terlihat?

Ini menjadi masalah bagi pengguna yang memang sejak awal ingin membedakan slug--misalnya saya ingin membuat slug lebih pendek--dari judul.

**Comment by taufik-nurrohman** [comment, marked as answer]
Biar lebih ringkas saja sih pak. Untuk pengguna awam, biasanya tidak begitu mengerti kegunaan bidang _slug_. Yang mereka tahu cuma soal formulir isian judul dan konten artikel. Bidang deskripsi kadang juga bisa membingungkan. Target pengguna kontrol panel sebenarnya adalah klien dari agensi yang membuat situs web menggunakan Mecha. Supaya agensi nanti nggak begitu kesulitan ketika harus menjelaskan cara kerja aplikasi kepada klien mereka.

Berikut ini cara mengaktifkan bidang _slug_ secara permanen:

``` php
Hook::set('_', function($_) {
    if (0 === strpos($_['type'] . '/', 'page/') && isset($_['lot']['desk']['lot']['form']['lot'][1]['lot']['tabs']['lot']['page']['lot']['fields']['lot']['name'])) {
        $_['lot']['desk']['lot']['form']['lot'][1]['lot']['tabs']['lot']['page']['lot']['fields']['lot']['name']['skip'] = false;
    }
    return $_;
});
```

Tanpa kode di atas nilai _slug_ sebenarnya masih bisa diganti setelah disimpan. Jadi urutannya: (1) Penulis membuat artikel dan menyimpannya sebagai draf. (2) Setelah itu penulis mengganti nilai _slug_ dan mempublikasikan draf artikel tersebut.
